src/cljdoc/doc_tree.clj

```diff
@@ -56,13 +58,15 @@
 
 ;; Specs for the Hiccup style configuration format that library authors
 ;; may use to specify articles and their hierarchy.
-(spec/def ::hiccup-attrs
-  (spec/keys :opt-un [::file]))
+(spec/def ::file-attrs
+  (spec/map-of #{:file} ::file))
+(spec/def ::url-attrs
+  (spec/map-of #{:url} :cljdoc.doc/external-url))
 
 (spec/def ::hiccup-entry
   (spec/spec
    (spec/cat :title ::title
-             :attrs (spec/? ::hiccup-attrs)
+             :attrs (spec/? (spec/or :file-attrs ::file-attrs :url-attrs ::url-attrs))
```

Comment:
I think we'd be better off going with this on line 61:

```clojure
(spec/def ::hiccup-attrs
  (spec/and (spec/keys :opt-un [::file ::url])
            #(not (and (:file %) (:url %)))))
```

And then on this line go back to using:

```clojure
(spec/? ::hiccup-attrs)
```

Then we could revert some of the changes below which would reduce a _lot_ of churn. That seems cleanest to me but I'm interested if there are other reasons this is better than I'm not seeing just yet.

Reply:
Maybe it wouldn't reduce churn at all and it still makes sense to break the file handling out into its own function? In which case this is fine as-is. I'll leave it up to you 🤷🏻‍♀️ .

Reply:
Since there's the other issue with `slug` down below, I think it's easier to keep the processing separated although it's a bigger change. 

Otherwise I'd need one more case to make sure the `slug` is not added which seems to be more coupling than necessary.